search: combine matches and progress

keegancsmith opened the pull request:
In the current protocol progress and matches events can come in at different times. This can lead to updating the list of results shown without updating progress. Or vice versa. To avoid that we group both matches and progress together.

lguychard reviewed (commented):
The direction feels right 👍

limitedmage commented:
I really like this! It makes a lot of sense. Let me know if you need any help implementing this refactor on the front end.

keegancsmith commented:
> I really like this! It makes a lot of sense. Let me know if you need any help implementing this refactor on the front end.

I'd love to pair on this with you and see how a TS dev does this change efficiently :) Especially including https://github.com/sourcegraph/sourcegraph/pull/16342 Can we pair around all the team meeting times on Monday? I'll prepare the backend changes so we should have something to work on end to end.

keegancsmith commented:
cc @stefanhengl

keegancsmith commented:
I just rebased this onto latest and updated the title/description. Once https://github.com/sourcegraph/sourcegraph/pull/17531 is ready this can land (with the appropriate API changes in the backend).

keegancsmith commented:
I don't think this is correct anymore.
